specification/kubernetesconfiguration/resource-manager/Microsoft.KubernetesConfiguration/preview/2021-05-01-preview/extensions.json

```diff
@@ -232,47 +248,87 @@
             "schema": {
-              "$ref": "#/definitions/ErrorResponse"
+              "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/ErrorResponse"
             }
           }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
         }
       }
     },
-    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/{clusterRp}/{clusterResourceName}/{clusterName}/providers/Microsoft.KubernetesConfiguration/extensions": {
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/{clusterRp}/{clusterResourceName}/{clusterName}/providers/Microsoft.KubernetesConfiguration/extensions/{extensionName}/operations": {
       "get": {
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ClusterRpParameter"
+          },
+          {
+            "$ref": "#/parameters/ClusterResourceNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ClusterNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ExtensionNameParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
         "tags": [
-          "Extensions",
-          "ClusterExtensions"
+          "Operations in a Cluster"
         ],
-        "description": "List all Source Control Configurations.",
-        "operationId": "Extensions_List",
+        "operationId": "OperationStatus_List",
+        "description": "List Async Operations, currently in progress, in a cluster",
         "x-ms-examples": {
-          "List Source Control Configuration": {
-            "$ref": "./examples/ListExtensions.json"
+          "AsyncOperationStatus List": {
+            "$ref": "./examples/ListAsyncOperationStatus.json"
           }
         },
+        "responses": {
+          "200": {
+            "description": "Operations in a cluster",
+            "schema": {
+              "$ref": "#/definitions/OperationStatusList"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/{clusterRp}/{clusterResourceName}/{clusterName}/providers/Microsoft.KubernetesConfiguration/extensions/{extensionName}/operations/{operationId}": {
```

Comment:
If I understand correctly, for an extension resource for which our RP received a DELETE request, once it is successfully deleted,  when ARM sends a GET on the OperationResult of this DELETE, if we return 404 for that request (since the extension resource itself is deleted and so not found), ARM will log this as Provider Error.  Is this correct?  If yes, in these cases, if we return 200OK, and properties.status=Deleted in the Response body, that should be sufficient, right?
To add more clarification, in all other cases - where the extension resource exists in our DB - we will have details of the operation and will be able to return a valid response to ARM for these OperationResult calls.

Reply:
Unfortunately, that won't work because only `Succeeded` , `Failed` and `Canceled` are considered terminal state. A `status` set to `Deleted` will be treated as a running operation by ARM, which will eventually fail due to the async operation timeout in ARM (and will generate provider errors).

You could potentially return `Canceled` in this case, but I can't promise it will work perfectly. Btw, the vast majority of RPs implement these at subscription and/or tenant scope. What's the reason for deviating?

Reply:
It's just that we have to create a separate collection in the DB and store this info, which then has to be kept in sync with the collection that contains the actual resources.  However, your point is valid regarding the Terminal States.  So, instead of Deleted, I'll return Succeeded then.

Reply:
I wouldn't try succeeded because there are cases where ARM will immediately query the resource state after `Succeeded`. If the resource is deleted, the result will be 404 and a different provider error.

Reply:
`Succeeded` is the correct state to report when the operation was the deletion of the resource, but `Canceled` should be reported for any other operation that was active when the resource got deleted.